### PR TITLE
Dynamic schema

### PR DESCRIPTION
**Pull request by vitaglianog**
This branch introduces a new module, called `SchemaBuilder`. The purpose of this module is to allow users to create palimpzest schemas dynamically from existing files.
The base syntax is the following 
```
schema = pz.SchemaBuilder.from_file(schema_path)
```

The builder can take as input a custom name, set of attributes to include, description and a type. 
It currently supports four file types: JSON, YAML, JSON-LD, and CSV.
- The most powerful readers are JSON and YAML, where users can fully specify information such as the parent class of the schema and the type of the inputs.
- The CSV reader is a convenience function that takes an existing csv and generates a schema that matches attribute names, and heuristically based on pandas detection of column types it casts the pz fields to different native field types.
- The JSON-LD reader is used to read schema files specified through json-ld formats (such as the ones distributed within synapse).

The module uses a fairly naive logic (the extension of the file) to check which function to call.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -78,6 +79,7 @@ namespaces = false
 
 
 [tool.pytest.ini_options]
+testpaths = ["tests/pytest"]
```
Comment (vitaglianog) [inline pyproject.toml:82]: This is used to call pytest from root dir and avoid the warning about `conftest.py`

File: testdata/images-tiny.zip
Comment (mdr223) [inline testdata/images-tiny.zip:1]: Is this meant to be included in the commit?

**Review by mdr223 — APPROVED** [review]
Left two minor comments; pre-approving as I think you can handle those and merge when ready.

File: testdata/images-tiny.zip
Comment (vitaglianog) [reply testdata/images-tiny.zip:1]: Oops nice catch!

**Pull request merged**